frappe/app.py

```diff
@@ -77,7 +77,7 @@ def application(request: Request):
 		rollback = after_request(rollback)
 
 	finally:
-		if request.method in ("POST", "PUT") and frappe.db and rollback:
+		if request.method in UNSAFE_HTTP_METHODS and frappe.db and rollback:
 			frappe.db.rollback()
```

Comment:
why is this bit needed at all if we're closing the db connection at the end of each request?

Reply:
Just safe transaction hygiene, we might chose to not close connections some day (pooling/persistence)